crates/sui-types/src/object.rs

```diff
@@ -414,12 +424,14 @@ impl Object {
     }
 
     /// Change the owner of `self` to `new_owner`
-    pub fn transfer(&mut self, new_owner: SuiAddress) -> SuiResult {
+    pub fn transfer(&mut self, new_owner: SuiAddress, should_update_version: bool) -> SuiResult {
```

Comment:
Nit: I think it would be more readable at call sites to expose two different functions rather than one with the boolean flag. I always find it difficult to remember what the boolean means, especially because Rust doesn't have named parameters.